fidesctl/Dockerfile

```diff
@@ -1,8 +1,12 @@
-FROM --platform=linux/amd64 python:3.8-slim-buster
+FROM --platform=linux/amd64 python:3.8-slim-buster as base
 
-######################
-# Tool Installation ##
-######################
+# Update pip in the base image since we'll use it everywhere
+RUN pip install -U pip
+
+FROM base as builder
+#######################
+## Tool Installation ##
+#######################
 
 RUN apt-get update
```

Comment:
A couple suggestions for your refactor.

By running `RUN apt-get update` on one line and `RUN apt-get install ...` on another, if you add new packages to install the `RUN apt-get update` layer will not run because it will be cached meaning you can miss security updates. By combining the update and install into the same lines you can avoid this. There are also a couple thing you can do here to reduce the overall size of the image.

```sh
 RUN : \
  && apt-get update \
  && apt-get install \
    -y --no-install-recommends \
    curl \
    git \
    ipython \
    make \
    vim \
    g++ \
    gnupg \
    gcc \
  && apt-get clean \
  && rm -rf /var/lib/apt/lists/*
```